### PR TITLE
Port uvloop to Python 3.12

### LINKED ISSUE #569: Can we update cython and make uvloop compatible with python 3.12 please?
* **uvloop version**: 0.17.0
* **Python version**: 3.12
* **Platform**: all
* **Can you reproduce the bug with `PYTHONASYNCIODEBUG` in env?**: N/A
* **Does uvloop behave differently from vanilla asyncio? How?**: N/A

Cython was recently updated and this allows uvloop to be compatible with python 3.12. There are a plethora of projects out there that relies on this. 

Is it possible to prioritize this and make a release, please? Asyncpg, for instance, it's one of them. 

Appreciate the understanding,
tarsil

### PR DESCRIPTION
**Pull request by 1st1**
Fixes #569

### CODE REVIEW

**Comment by 1st1** [comment]
While working on this I stumbled upon what seems to be an asyncio regression in 3.12. Good news that uvloop doesn't have this regression. Bad news -- our tests that test compliance with asyncio are now failing (we should backport our functional tests to the Python project, they increase test coverage by a lot!)

File: setup.py [hunk 1]
```diff
@@ -37,6 +36,15 @@
     CYTHON_DEPENDENCY,
 ]
 
+if vi < (3, 12):
+    # XXX Revert this change later.
+    #
+    # Python 3.12 is new and there's no aiohttp wheel for it yet.
+    # And pip helfully fails on building a wheel for it and I've
+    # no idea how to disable that.
```
Comment (graingert) [inline setup.py:44]: There's an environment variable you can set to use the pure Python http parser

File: README.rst [hunk 2]
```diff
@@ -53,6 +53,27 @@ uvloop with::
 Using uvloop
 ------------
 
+Starting with Python 3.12, the preferred way of using uvloop
```
Comment (graingert) [inline README.rst:56]: uvloop.run seems to be available on all supported Python versions so I think this documentation should be "as of uvloop x.y, the preferred way of..."

**Review by graingert — CHANGES_REQUESTED** [review]

File: README.rst (hunk 2 above)
Comment (1st1) [reply README.rst:56]: good one, thanks

File: setup.py (hunk 1 above)
Comment (1st1) [reply setup.py:44]: The problem isn't using aiohttp, it's about not being able to install it with 3.12 in CI

Comment (graingert) [reply setup.py:44]: I know, the environment variable will let you install it in CI

Comment (graingert) [reply setup.py:44]: Add this to your workflow and the aiohttp wheel will build and install from pypi

```
      - name: Set AIOHTTP_NO_EXTENSIONS=1 on 3.12
        if: matrix.environment == '3.12'
        run: echo AIOHTTP_NO_EXTENSIONS=1 >> $GITHUB_ENV
        shell: bash
```

Comment (fantix) [reply setup.py]: Hmm, actually, I'll just use `aiohttp== 3.9.0b0`, which was only released 2 days ago.

**Review by fantix — APPROVED** [review]

**Comment by AntonOfTheWoods** [comment]
Is there an ETA on a release?

**Pull request merged**